ios/HeroNotifications/NotificationViewController.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  NotificationViewController.swift
+//  HeroNotifications
+//
+//  Created by Francisco Jordano on 17/11/2016.
+//  Copyright © 2016 Mozilla. All rights reserved.
+//
+
+import UIKit
+import UserNotifications
+import UserNotificationsUI
+
+class NotificationViewController: UIViewController, UNNotificationContentExtension {
+
+    @IBOutlet var imageView: UIImageView!
+  
+    private static var notification: UNNotification?
+  
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+  
+    func didReceiveNotification(notification: UNNotification) {
```

Comment:
It's not magic is configured by xcode, into the xmls it says that is the controller for the view that will be displayed, in this case the view for the extended notification. But yes, happens when you initialize the template in xcode.